cbits/cryptonite_aes.c

```diff
@@ -406,6 +438,139 @@ void cryptonite_aes_gcm_finish(uint8_t *tag, aes_gcm *gcm, aes_key *key)
 	}
 }
 
+static inline int ccm_b0_flags(int has_adata, int m, int l)
+{
+	return 8*m + l + (has_adata? 64: 0);
+}
+
+/* depends on input size */
+static void ccm_encode_b0(block128* output, aes_ccm* ccm, int has_adata)
+{
+	int last = 15;
+	int m = ccm->length_M;
+	int l = ccm->length_L;
+	unsigned msg_len = ccm->length_input;
+
+	block128_zero(output);
+	block128_copy(output, &ccm->nonce);
+	output->b[0] = ccm_b0_flags(has_adata, (m-2)/2, l-1);
+	while (msg_len > 0) {
+		output->b[last--] = msg_len & 0xff;
+		msg_len >>= 8;
+	}
+}
+
+/* encode adata length */
+static int ccm_encode_la(block128* output, unsigned la)
+{
+	if (la < ( (1 << 16) - (1 << 8)) ) {
+		output->b[0] = (la >> 8) & 0xff;
+		output->b[1] = la        & 0xff;
+		return 2;
+	} else if (la < (1ull << 32)) {
+		output->b[0] = 0xff;
+		output->b[1] = 0xfe;
+		output->b[2] = (la >> 24) & 0xff;
+		output->b[3] = (la >> 16) & 0xff;
+		output->b[4] = (la >>  8) & 0xff;
+		output->b[5] = la         & 0xff;
+		return 6;
+	}
+}
```

Comment:
With clang I get two warnings related to function `ccm_encode_la`:
```
cbits/cryptonite_aes.c:470:16: error:
     warning: comparison of constant 4294967296 with expression of type 'unsigned int' is always true [-Wtautological-constant-out-of-range-compare]
            } else if (la < (1ull << 32)) {
                       ~~ ^ ~~~~~~~~~~~~
    |
470 |         } else if (la < (1ull << 32)) {
    |                ^

cbits/cryptonite_aes.c:479:1: error:
     warning: control may reach end of non-void function [-Wreturn-type]
    |
479 | }
    | ^
}
^
2 warnings generated.
```

Can variable `la` be changed to `uint32_t` (like in the caller) and the test removed?